week-4/implement/count.test.js

```diff
@@ -1,5 +1,24 @@
 // implement a function countChar that counts the number of times a character occurs in a string
 
+function countChar(str, char) {
+  let index = 0;
+  for (let i = 0; i < str.length; i++) {
+    if (str[i] === char) {
+      index++;
+    }
+    while (str[i + 1] === char) {
```

Comment:
From my understanding, this loop will keep counting up and moving through the string if the next element in the string is the character we're counting. However if we just let the `for` loop run, it will already check every character in the string. So this `while` loop isn't doing much for us, just using the `for` loop will be easier to understand 🙂